### ISSUE TITLE
CIFAR-10 Evaluation

### ISSUE THREAD

**Issue by harveymannering** [atong01/conditional-flow-matching#176]
Thanks for this codebase, it is super useful!

I have a question about the CIFAR-10 FID evaluation. If you look at the [this line](https://github.com/atong01/conditional-flow-matching/blob/main/examples/images/cifar10/compute_fid.py#L101), the FID evaluation for CFM is done using the train split of CIFAR-10. Shouldn't this be done using the test set? Or does it not matter? If not, why not?

**Comment by Charley-xiao** [comment]
Hey @harveymannering,

It's a common practice in generative modeling, especially GAN and diffusion, to compute FID on the training set. For example, in [DDPM](https://arxiv.org/pdf/2006.11239), "Our FID score is computed with respect to the training set, as is standard practice; when we compute it with respect to the test set, the score is 5.24..."

However, there are papers like [this one](https://proceedings.neurips.cc/paper_files/paper/2023/file/68b138608ef80b08d65b1bd9594d9559-Paper-Conference.pdf) that argue train-FID cannot represent generalizability, so you'll also see people computing FID on the test set in tasks like image-to-image translation. So it really depends on how previous works were doing it.

Hope that helps!
